### PR TITLE
Create initial layout of profile page

### PR DESCRIPTION
**Pull request by Luiz-FS**
**Add profile page of logged user.**
* Have a profile screen.
* Have a follower screen.

diagrama_task

### CODE REVIEW

File: app/src/main/java/com/github/ghcli/models/UserGitModel.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
+package com.github.ghcli.models;
+
+/**
```
Comment (llvieira) [inline app/src/main/java/com/github/ghcli/models/UserGitModel.java]: Remove this comment.

File: app/src/main/java/com/github/ghcli/models/UserGitModel.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+ * Created by davi on 14/11/2017.
+ */
+
+public class UserGitModel {
+
+    private String login;
+    private String name;
+    private String type;
+    private String avatar_url;
```
Comment (llvieira) [inline app/src/main/java/com/github/ghcli/models/UserGitModel.java]: `avatar_url` is not a Java pattern to variable names. You should do:
```java
import com.google.gson.annotations.SerializedName;

@SerializedName("avatar_url")
private String avatarUrl;
```

File: app/src/main/java/com/github/ghcli/models/UserGitModel.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
+package com.github.ghcli.models;
+
+/**
+ * Created by davi on 14/11/2017.
+ */
+
+public class UserGitModel {
```
Comment (llvieira) [inline app/src/main/java/com/github/ghcli/models/UserGitModel.java]: A better name to this class could be `GitHubUser`. There's no need to write `Model` in classes name.

File: app/src/main/java/com/github/ghcli/models/UserGitModel.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+    public String getType() {
+        return type;
+    }
+
+    public void setType(String type) {
+        this.type = type;
+    }
+
+    public String getAvatar_url() {
```
Comment (llvieira) [inline app/src/main/java/com/github/ghcli/models/UserGitModel.java]: After the fix above, change here.

File: app/src/main/java/com/github/ghcli/models/UserGitModel.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        return avatar_url;
+    }
+
+    public void setAvatar_url(String avatar_url) {
```
Comment (llvieira) [inline app/src/main/java/com/github/ghcli/models/UserGitModel.java]: After the fix above, change here.

File: app/src/main/java/com/github/ghcli/services/GitService.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
+package com.github.ghcli.services;
+
+import com.github.ghcli.models.UserGitModel;
+
+import retrofit2.Call;
+import retrofit2.http.GET;
+import retrofit2.http.Path;
+
+/**
```
Comment (llvieira) [inline app/src/main/java/com/github/ghcli/services/GitService.java]: Remove this comment.

File: app/src/main/java/com/github/ghcli/services/GitService.java [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+ * Created by davi on 14/11/2017.
+ */
+
+public interface GitService {
+
+    public static final String BASE_URL = "https://api.github.com/";
```
Comment (llvieira) [inline app/src/main/java/com/github/ghcli/services/GitService.java]: This constant already exists in `ServiceGenerator` class. No longer need this here.

File: app/src/main/java/com/github/ghcli/services/GitService.java [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+    @GET("users/{username}")
+    Call<UserGitModel> getUserGit(@Path("username") String username);
```
Comment (llvieira) [inline app/src/main/java/com/github/ghcli/services/GitService.java]: Rename to `getGitHubUser`.

**Comment by GabrielSVinha** [comment]
@Luiz-FS @liraop This PR contains 31 commits, please squash them into at maximum 4.
Here is some guidance: https://imasters.com.br/desenvolvimento/como-fazer-squash-de-commits-no-git/?trace=1519021197&source=single

File: app/src/main/java/com/github/ghcli/ViewHolder/FollowersViewHolder.java [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+            }
+
+            @Override
+            public void onFailure(Call<Void> call, Throwable t) {
+            }
+        });
+    }
+
+    private void setActionButton() {
+        this.follow.setOnClickListener(new View.OnClickListener() {
```
Comment (llvieira) [inline app/src/main/java/com/github/ghcli/ViewHolder/FollowersViewHolder.java]: You should use the **ButterKnife** here.

```java
@OnClick(R.id.<button_id>)
private void setActionButton() {
     if (following) {
        ....
    }
}
```

File: app/src/main/java/com/github/ghcli/ViewHolder/FollowersViewHolder.java [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+    private View view;
+    private Context context;
+    private IGitHubUser iGitHubUser;
+    private boolean following;
+
+    public FollowersViewHolder(View itemView, IGitHubUser iGitHubUser, Context context) {
+        super(itemView);
+        this.login = (TextView) itemView.findViewById(R.id.follower_name);
+        this.follow = (Button) itemView.findViewById(R.id.follow);
+        this.avatar = (ImageView) itemView.findViewById(R.id.follower_avatar);
```
Comment (llvieira) [inline app/src/main/java/com/github/ghcli/ViewHolder/FollowersViewHolder.java]: To use **ButterKnife** to login, follow and avatar.

Comment (llvieira) [reply app/src/main/java/com/github/ghcli/ViewHolder/FollowersViewHolder.java]: You should remember that to use **ButterKnife** the attribute needs to be `public`.

**Pull request merged**